File: client/bcephal-client-model/Settings/ParameterGroup.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bcephal.Models.Settings
{
    public class ParameterGroup
    {

        public string Code { get; set; }

        public string Name { get; set; }

        public bool CanBeCreateAutomatically { get; set; }

        public ObservableCollection<ParameterGroupItem> ParameterGroupItems { get; set; }

        public ObservableCollection<ParameterGroup> SubGroups { get; set; }


        public ParameterGroup()
        {
            this.ParameterGroupItems = new ObservableCollection<ParameterGroupItem>();
            this.SubGroups = new ObservableCollection<ParameterGroup>();
            this.CanBeCreateAutomatically = false;
        }

        public ParameterGroup(String code) : this()
        {
            this.Code = code;
        }

        public ParameterGroup(String code, string name)
            : this(code)
        {
            this.Name = name;
        }

        public ParameterGroup(String code, string name, bool canBeCreateAutomatically)
            : this(code, name)
        {
            this.CanBeCreateAutomatically = canBeCreateAutomatically;
        }


        public override string ToString()
        {
            return this.Name;
        }

    }
}

File: server/bcephal-settings-service/src/main/java/com/moriset/bcephal/settings/domain/ParameterGroup.java
/**
 * 
 */
package com.moriset.bcephal.settings.domain;

import java.util.ArrayList;
import java.util.List;

import lombok.Data;

/**
 * @author MORISET-6
 *
 */
@Data
public class ParameterGroup {

	private String code;

	private String name;
	
	private Boolean canBeCreateAutomatically;
	
	private List<ParameterGroupItem> ParameterGroupItems;
	
	public List<ParameterGroup> subGroups ;
	
	public ParameterGroup() {
		subGroups = new ArrayList<ParameterGroup>(0);
		ParameterGroupItems = new ArrayList<ParameterGroupItem>(0);
	}
	
	public ParameterGroup(String code) {
		this();
		this.code = code;
	}
	
	public ParameterGroup(String code, String name) {
		this();
		this.code = code;
		this.name = name;
	}
	
	public ParameterGroup(String code, String name, Boolean canBeCreateAutomatically) {
		this();
		this.code = code;
		this.name = name;
		this.canBeCreateAutomatically = canBeCreateAutomatically;
	}

}
